tests/unit/test_resultset.py

```
@@ -175,11 +175,18 @@ def test_bool(self):
     def test_was_applied(self):
+        # Create a non-LWT query so these assertions exercise the slow (regex) path.
+        # Without this, Mock().query.is_lwt() returns a truthy Mock, accidentally
+        # routing all checks through the fast path.
+        non_lwt_query = Mock(spec=SimpleStatement)
+        non_lwt_query.is_lwt.return_value = False
+        non_lwt_query.query_string = "INSERT INTO t (k) VALUES (1)"
+
         # unknown row factory raises
         with pytest.raises(RuntimeError):
-            ResultSet(Mock(), []).was_applied
+            ResultSet(Mock(query=non_lwt_query), []).was_applied
```

Ruff (0.16.1)

[warning] Found useless attribute access. Either assign it to a variable or remove it.

(B018)